--- ltrace-0.3.32/output.c.orig	2003-02-02 22:51:33.000000000 +0100
+++ ltrace-0.3.32/output.c	2004-04-18 18:40:05.479364624 +0200
@@ -88,11 +88,11 @@
 	}
 	if (opt_i) {
 		if (type==LT_TOF_FUNCTION) {
-			current_column += fprintf(output, "[%08x] ",
-				(unsigned)proc->return_addr);
+			current_column += fprintf(output, "[%p] ",
+				proc->return_addr);
 		} else {
-			current_column += fprintf(output, "[%08x] ",
-				(unsigned)proc->instruction_pointer);
+			current_column += fprintf(output, "[%p] ",
+				proc->instruction_pointer);
 		}
 	}
 	if (opt_n > 0 && type!=LT_TOF_NONE) {
--- ltrace-0.3.32/display_args.c.orig	2004-04-04 00:37:00.000000000 +0200
+++ ltrace-0.3.32/display_args.c	2004-04-18 18:49:44.567329872 +0200
@@ -39,7 +39,7 @@
 			if (!arg) {
 				return fprintf(output, "NULL");
 			} else {
-				return fprintf(output, "%p", (unsigned)arg);
+				return fprintf(output, "%p", (void*)arg);
 			}
 		case ARGTYPE_FORMAT:
 			return display_format(type, proc, arg_num);
