--- ltrace-0.5.orig/options.h
+++ ltrace-0.5/options.h
@@ -20,6 +20,7 @@
 extern int opt_C;		/* Demanglelow-level symbol names into user-level names */
 extern int opt_n;		/* indent trace output according to program flow */
 extern int opt_T;		/* show the time spent inside each call */
+extern int opt_o;		/* output to a specific file */
 
 struct opt_p_t {
 	pid_t pid;
--- ltrace-0.5.orig/options.c
+++ ltrace-0.5/options.c
@@ -42,6 +42,7 @@
 #endif
 int opt_n = 0;			/* indent trace output according to program flow */
 int opt_T = 0;			/* show the time spent inside each call */
+int opt_o = 0;			/* output to a specific file */
 
 /* List of pids given to option -p: */
 struct opt_p_t *opt_p = NULL;	/* attach to process with a given pid */
@@ -274,6 +275,7 @@
 			options.indent = atoi(optarg);
 			break;
 		case 'o':
+			opt_o++;
 			options.output = fopen(optarg, "w");
 			if (!options.output) {
 				fprintf(stderr,
--- ltrace-0.5.orig/libltrace.c
+++ ltrace-0.5/libltrace.c
@@ -54,6 +54,9 @@
 {
 	exiting = 1;
 	debug(1, "Received interrupt signal; exiting...");
+	if (opt_o) {
+		fclose(options.output);
+	}
 	signal(SIGINT, SIG_IGN);
 	signal(SIGTERM, SIG_IGN);
 	signal(SIGALRM, signal_alarm);
