--- ltrace-0.3.26/Makefile.in	Mon Apr  1 09:42:30 2002
+++ ltrace-0.3.26/Makefile.in.orig	Mon Apr  1 09:47:10 2002
@@ -16,9 +16,14 @@
 CC	=	@CC@
 CPPFLAGS =	-I$(TOPDIR) -I$(TOPDIR)/sysdeps/$(OS) -I- @DEFS@
 CFLAGS	=	-Wall @CFLAGS@ -DSYSCONFDIR=\"$(sysconfdir)\"
-LDFLAGS	=
+LDFLAGS	=       @LDFLAGS@
 LIBS	=	@LIBS@
 
+INSTALL =      @INSTALL@
+mandir =       @mandir@
+bindir =       @bindir@
+
+
 OBJ	=	ltrace.o options.o elf.o output.o read_config_file.o	\
 		execute_program.o wait_for_something.o process_event.o	\
 		display_args.o breakpoints.o proc.o demangle.o
@@ -46,12 +51,11 @@
 		( cd .. ; tar zcvf ltrace-`date +%y%m%d`.tgz ltrace )
 
 install:	ltrace
-		@INSTALL@ -d $(DESTDIR)$(bindir) $(DESTDIR)$(docdir) $(DESTDIR)$(mandir)/man1
-		@INSTALL@ -d $(DESTDIR)$(sysconfdir)
-		@INSTALL@ -s ltrace $(DESTDIR)$(bindir)
-		@INSTALL@ -m 644 etc/ltrace.conf $(DESTDIR)$(sysconfdir)
-		@INSTALL@ -m 644 COPYING README TODO BUGS ChangeLog $(DESTDIR)$(docdir)
-		@INSTALL@ -m 644 ltrace.1 $(DESTDIR)$(mandir)/man1
+		$(INSTALL) -d $(DESTDIR)$(bindir) $(DESTDIR)$(mandir)/man1		
+		$(INSTALL) -d $(DESTDIR)/etc
+		$(INSTALL) ltrace $(DESTDIR)$(bindir)
+		$(INSTALL) -m 644 etc/ltrace.conf $(DESTDIR)/etc
+		$(INSTALL) -m 644 ltrace.1 $(DESTDIR)$(mandir)/man1
 
 dummy:
 
